walnuts_cpp/include/walnuts/nuts.hpp

```diff
@@ -98,9 +100,19 @@ void leapfrog(const F &logp_grad_fun, const Vec<S> &inv_mass, S step,
   logp_next += logp_momentum(rho_next, inv_mass);
 }
 
-template <typename S>
-bool uturn(const Span<S> &span_bk, const Span<S> &span_fw,
-           const Vec<S> &inv_mass) {
+template <Direction D, typename S>
+inline auto order_forward_backward(S &&s1, S &&s2) {
+  if constexpr (D == Direction::Forward) {
+    return std::forward_as_tuple(std::forward<S>(s1), std::forward<S>(s2));
+  } else { // Direction::Backward
+    return std::forward_as_tuple(std::forward<S>(s2), std::forward<S>(s1));
+  }
+}
```

Comment:
Note this returns a `std::tuple<Span<S>&&, Span<S>&&>` type. This works because s1 and s2 are inside of the scope in the outer function, but if this was called with either s1 or s2 would be destroyed once this function ended. this works because of how it is used, but we should be careful in the future with this.